core/src/main/java/org/mskcc/cbio/portal/dao/DaoGeneset.java

```diff
@@ -0,0 +1,466 @@
+            List<CanonicalGene> genes = new ArrayList<CanonicalGene>();
+            for (Long entrezGeneId : entrezGeneIds) {
+                CanonicalGene gene = daoGeneOptimized.getGene(entrezGeneId);
+                genes.add(gene);
+            }
+            
+            return genes;
+        }
+        catch (SQLException e) {
+            throw new DaoException(e);
+        } 
+        finally {
+            JdbcUtil.closeAll(DaoGeneset.class, con, pstmt, rs);
+        }
+    }
+    
+    
+    /**
+     * Given an external id, returns a Geneset record.
+     * @param externalId
+     * @return Geneset record
+     * @throws DaoException 
+     */
+    public static Geneset getGenesetByExternalId(String externalId) throws DaoException {
+        Connection con = null;
+        PreparedStatement pstmt = null;
+        ResultSet rs = null;        
+        try {
+            con = JdbcUtil.getDbConnection(DaoGeneset.class);
+            pstmt = con.prepareStatement("SELECT * FROM geneset WHERE `EXTERNAL_ID` = ?");
+            pstmt.setString(1, externalId);
+            rs = pstmt.executeQuery();
+            
+            // return null if result set is empty
+            if (rs.next()) {
+                return extractGeneset(rs);
+            }
+            else {
+                return null;
+            }
+        }
+        catch (SQLException e) {
+            throw new DaoException(e);
+        } 
+        finally {
+            JdbcUtil.closeAll(DaoGeneset.class, con, pstmt, rs);
+        }
+    }
+
+    
+    /**
+     * Get Geneset record.
+     */
+    public Geneset getGenesetById(int genesetId) throws DaoException {
+        Connection con = null;
+        PreparedStatement pstmt = null;
+        ResultSet rs = null;
+        try {
+            con = JdbcUtil.getDbConnection(DaoGeneset.class);
+            pstmt = con.prepareStatement("SELECT * FROM geneset WHERE ID = ?");
+            pstmt.setInt(1, genesetId);
+            System.out.println(pstmt);
+            rs = pstmt.executeQuery();
+            if (rs.next()) {
+                //return extractGene(rs);
+            	return extractGeneset(rs);
+            } else {
+                return null;
+            }
+        } catch (SQLException e) {
+            throw new DaoException(e);
+        } finally {
+            JdbcUtil.closeAll(DaoGeneset.class, con, pstmt, rs);
+        }
+    }
+
+    
+    /**
+     * Extracts Geneset record from ResultSet.
+     * @param rs
+     * @return Geneset record
+     * @throws SQLException
+     * @throws DaoException 
+     */
+    private static Geneset extractGeneset(ResultSet rs) throws SQLException, DaoException {
+        Integer id = rs.getInt("ID");
+        Integer geneticEntityId = rs.getInt("GENETIC_ENTITY_ID");
+        String externalId = rs.getString("EXTERNAL_ID");
+        String name = rs.getString("NAME");
+        String description = rs.getString("DESCRIPTION");
```

Comment:
The version of the `geneset` table that I have in my own test database contains the following columns:

```
+-------------------+--------------+------+-----+---------+-------+
| Field             | Type         | Null | Key | Default | Extra |
+-------------------+--------------+------+-----+---------+-------+
| ID                | int(11)      | NO   | PRI | NULL    |       |
| GENETIC_ENTITY_ID | int(11)      | NO   | UNI | NULL    |       |
| EXTERNAL_ID       | varchar(100) | NO   | UNI | NULL    |       |
| NAME_SHORT        | varchar(100) | NO   | UNI | NULL    |       |
| NAME              | varchar(300) | NO   |     | NULL    |       |
| REF_LINK          | text         | YES  |     | NULL    |       |
| VERSION           | varchar(50)  | NO   |     | NULL    |       |
+-------------------+--------------+------+-----+---------+-------+
```

There's no `DESCRIPTION` column in my version of this table... whatever version of these tables that I have are from back in December when I first started working with @pieterlukasse on supporting GSVA data loading so the result set can't be parsed correctly here. Was this column added in a subsequent version of these geneset-related tables? Where can we find the changes to `cgds.sql` and `migration.sql`? @pieterlukasse @sandertan